### ISSUE TITLE
is "Project loading failed: (r)etry" getting stuck on CI?

### ISSUE THREAD

**Issue by eed3si9n** [sbt/sbt#5790]
Ref https://github.com/sbt/sbt/issues/5624

## steps

Run Community Build (apparently it's now singular) using sbt 1.4.0-M2.

## problem

The first loading error it encounters the entire thing times out.

https://scala-ci.typesafe.com/job/scala-2.13.x-jdk11-integrate-community-build/1781/console

```
[akka-http] [error] ${BASE}/project/ValidatePullRequest.scala:376:62: type mismatch;
[akka-http] [error]  found   : (sbt.internal.util.Appender, sbt.Level.Value)
[akka-http] [error]  required: (org.apache.logging.log4j.core.Appender, sbt.util.Level.Value)
[akka-http] [error]       LogExchange.bindLoggerAppenders("testLogger", appender -> Level.Info :: Nil)
[akka-http] [error]                                                              ^
[akka-http] [error] one error found
[akka-http] [error] Compilation failed
[akka-http] [error] 	at xsbt.CachedCompiler0.handleErrors(CompilerBridge.scala:183)
[akka-http] [error] 	at xsbt.CachedCompiler0.run(CompilerBridge.scala:172)
[akka-http] [error] 	at xsbt.CachedCompiler0.run(CompilerBridge.scala:134)
[akka-http] [error] 	at xsbt.CompilerBridge.run(CompilerBridge.scala:39)
[akka-http] [error] 	at sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:89)
[akka-http] [error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$7(MixedAnalyzingCompiler.scala:182)
[akka-http] [error] 	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
[akka-http] [error] 	at sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:237)
[akka-http] [error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:172)
[akka-http] [error] 	at sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4$adapted(MixedAnalyzingCompiler.scala:153)
[akka-http] [error] 	at sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:233)
[akka-http] [error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:153)
[akka-http] [error] 	at sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:200)
[akka-http] [error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:571)
[akka-http] [error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:571)
[akka-http] [error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5(Incremental.scala:175)
[akka-http] [error] 	at sbt.internal.inc.Incremental$.$anonfun$apply$5$adapted(Incremental.scala:173)
[akka-http] [error] 	at sbt.internal.inc.Incremental$$anon$2.run(Incremental.scala:446)
[akka-http] [error] 	at sbt.internal.inc.IncrementalCommon$CycleState.next(IncrementalCommon.scala:121)
[akka-http] [error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:56)
[akka-http] [error] 	at sbt.internal.inc.IncrementalCommon$$anon$1.next(IncrementalCommon.scala:52)
[akka-http] [error] 	at sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:288)
[akka-http] [error] 	at sbt.internal.inc.Incremental$.$anonfun$incrementalCompile$7(Incremental.scala:413)
[akka-http] [error] 	at sbt.internal.inc.Incremental$.withClassfileManager(Incremental.scala:486)
[akka-http] [error] 	at sbt.internal.inc.Incremental$.incrementalCompile(Incremental.scala:400)
[akka-http] [error] 	at sbt.internal.inc.Incremental$.apply(Incremental.scala:167)
[akka-http] [error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:571)
[akka-http] [error] 	at sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:489)
[akka-http] [error] 	at sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:332)
[akka-http] [error] 	at sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:419)
[akka-http] [error] 	at sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:137)
[akka-http] [error] 	at sbt.Defaults$.compileIncrementalTaskImpl(Defaults.scala:2038)
[akka-http] [error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$2(Defaults.scala:2004)
[akka-http] [error] 	at sbt.internal.io.Retry$.apply(Retry.scala:40)
[akka-http] [error] 	at sbt.internal.io.Retry$.apply(Retry.scala:23)
[akka-http] [error] 	at sbt.internal.server.BspCompileTask$.compute(BspCompileTask.scala:31)
[akka-http] [error] 	at sbt.Defaults$.$anonfun$compileIncrementalTask$1(Defaults.scala:2001)
[akka-http] [error] 	at scala.Function1.$anonfun$compose$1(Function1.scala:49)
[akka-http] [error] 	at sbt.internal.util.$tilde$greater.$anonfun$$u2219$1(TypeFunctions.scala:62)
[akka-http] [error] 	at sbt.std.Transform$$anon$4.work(Transform.scala:68)
[akka-http] [error] 	at sbt.Execute.$anonfun$submit$2(Execute.scala:282)
[akka-http] [error] 	at sbt.internal.util.ErrorHandling$.wideConvert(ErrorHandling.scala:23)
[akka-http] [error] 	at sbt.Execute.work(Execute.scala:291)
[akka-http] [error] 	at sbt.Execute.$anonfun$submit$1(Execute.scala:282)
[akka-http] [error] 	at sbt.ConcurrentRestrictions$$anon$4.$anonfun$submitValid$1(ConcurrentRestrictions.scala:233)
[akka-http] [error] 	at sbt.CompletionService$$anon$2.call(CompletionService.scala:59)
[akka-http] [error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[akka-http] [error] 	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[akka-http] [error] 	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[akka-http] [error] 	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[akka-http] [error] 	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[akka-http] [error] 	at java.base/java.lang.Thread.run(Thread.java:834)
[akka-http] [error] (Compile / compileIncremental) Compilation failed
[akka-http] [warn] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? (default: r)
Build timed out (after 15 minutes). Marking the build as aborted.
Build was aborted
```

## expectation

On CI (or whenever standard input is not attached), don't wait for the user to enter `r` etc.

Previously even when one build has a problem loading, Community Build was able to move on to another build:

```
[akka-management] [error] ${BASE}/project/Common.scala:50:19: type mismatch;
[akka-management] [error]  found   : Seq[String]
[akka-management] [error]  required: Seq[Nothing]
[akka-management] [error]           else Seq("--release", "8")
[akka-management] [error]                   ^
[akka-management] [error] ${BASE}/project/Common.scala:48:20: type mismatch;
[akka-management] [error]  found   : sbt.Def.Initialize[sbt.Task[Seq[Nothing]]]
[akka-management] [error]  required: sbt.Def.Initialize[sbt.Task[Seq[String]]]
[akka-management] [error]       javacOptions ++= (
[akka-management] [error]                    ^
[akka-management] [error] two errors found
[akka-management] [error] (Compile / compileIncremental) Compilation failed
[akka-management] [warn] Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore? (default: r)
[akka-management:error] java.lang.RuntimeException: 
[akka-management:error] 	at scala.sys.package$.error(package.scala:27)
[akka-management:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner.com$typesafe$dbuild$support$sbt$SbtRunner$$processCommand(SbtRunner.scala:94)
[akka-management:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1$$anonfun$apply$2$$anonfun$apply$3.apply(SbtRunner.scala:75)
[akka-management:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1$$anonfun$apply$2$$anonfun$apply$3.apply(SbtRunner.scala:75)
[akka-management:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1.apply(SbtRunner.scala:75)
[akka-management:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner$$anonfun$run$1.apply(SbtRunner.scala:66)
[akka-management:error] 	at sbt.IO$.withTemporaryFile(IO.scala:389)
[akka-management:error] 	at com.typesafe.dbuild.support.sbt.SbtRunner.run(SbtRunner.scala:66)
[akka-management:error] 	at com.typesafe.dbuild.support.sbt.SbtExtractor$.extractMetaData(DependencyExtractor.scala:107)
[akka-management:error] 	at com.typesafe.dbuild.support.sbt.SbtBuildSystem.extractDependencies(SbtBuildSystem.scala:85)
[akka-management:error] 	at com.typesafe.dbuild.support.sbt.SbtBuildSystem.extractDependencies(SbtBuildSystem.scala:22)
[akka-management] --== End Extracting dependencies for akka-management ==--
[akka-more] --== Extracting dependencies for akka-more ==--
[akka-more] Fetching https://github.com/akka/akka.git
...
```

**Comment by eatkins** [comment]
Looks like there is also a binary compatibility issue with the logging change that I should try to resolve.

**Comment by eatkins** [comment]
The project load failing issue may also be related to #5788.

**Comment by eatkins** [comment]
I verified locally that the hangs are fixed using the latest sbt snapshot.

**Comment by eatkins** [comment]
I also fixed the cause of the hang in #5807.
